simplecpp.cpp

```diff
@@ -3257,12 +3264,20 @@ void simplecpp::preprocess(simplecpp::TokenList &output, const simplecpp::TokenL
         for (simplecpp::MacroMap::const_iterator macroIt = macros.begin(); macroIt != macros.end(); ++macroIt) {
             const Macro &macro = macroIt->second;
             const std::list<Location> &usage = macro.usage();
-            for (std::list<Location>::const_iterator usageIt = usage.begin(); usageIt != usage.end(); ++usageIt) {
-                MacroUsage mu(usageIt->files, macro.valueDefinedInCode());
-                mu.macroName = macro.name();
-                mu.macroLocation = macro.defineLocation();
-                mu.useLocation = *usageIt;
-                macroUsage->push_back(mu);
+            std::list<std::list<Location> > usageLists;
```

Comment:
hmm.. an alternative could be to append the two lists into one list..

    std::list<Location> usage = macro.usage();
    const std::list<Location>& temp = maybeUsedMacros[macro.name()];
    usage.insert(usage.end(), temp.begin(), temp.end());

Reply:
I was trying to avoid deep copies, but agree the code became too convoluted. Once again my C++ inexperience is showing. I'll apply your suggestion.

Reply:
Force pushed!